### PR TITLE
Fix unload in sshkey client #173

### PR DESCRIPTION
**Pull request by ranatrk**
Issue: #173 
When unloading sshkey from sshagent, it is unloaded from the agent but an exception is thrown.
Fix: 
- fixes in  is_loaded() check
- fixes in unload()
```
sshkey_client = j.clients.sshkey.get(name=SSHKEYCLIENT_NAME, passphrase_=PASSPHRASE)
sshkey_client.load()
sshkey_client.unload()
```

### CODE REVIEW

File: JumpscaleCore/clients/sshkey/SSHKey.py [hunk 1]
```diff
@@ -144,12 +144,12 @@ def load(self):
     def unload(self):
         cmd = "ssh-add -d %s " % (self.path)
         rc = 0
-        while rc == 0:
+        err = ""
+        while rc == 0 and ("Identity removed" not in err):
```
Comment (ahussein) [inline JumpscaleCore/clients/sshkey/SSHKey.py]: I am not sure how keep looping and executing the same command will help?

File: JumpscaleCore/clients/sshkey/SSHKey.py [hunk 2]
```diff
@@ -159,12 +159,13 @@ def is_loaded(self):
         :return: whether ssh key was loadeed in ssh agent or not
         :rtype: bool
         """
+        curr_pubkey = self.pubkey.split(" ")[1].strip()
```
Comment (ahussein) [inline JumpscaleCore/clients/sshkey/SSHKey.py]: this is not needed, there is a property called `pubkey_only` in the ssh client object.

**Review by ahussein — COMMENTED** [review]
I am already working on some changes to fix the sshkey failing tests. The changes already cover the problems this PR is intended to fix

**Comment by ahussein** [comment]
related PR here: https://github.com/threefoldtech/jumpscaleX_core/pull/188

File: JumpscaleCore/clients/sshkey/SSHKey.py (hunk 2 above)
Comment (ranatrk) [reply JumpscaleCore/clients/sshkey/SSHKey.py]: This is sshkey client not ssh client so `pubkey_only` should not be available here right?

Comment (ranatrk) [reply JumpscaleCore/clients/sshkey/SSHKey.py]: yes `pubkey_only` should be used instead

**Pull request closed without merging**